### PR TITLE
Inplace computations and binary ops

### PR DESCRIPTION
**Pull request by pgovind**
Making the API more consistent around whether it operates in place or not. Should be an easy review.

### CODE REVIEW

File: src/Microsoft.Data/PrimitiveColumn.BinaryOperations.tt [hunk 1]
```diff
@@ -77,7 +101,11 @@ namespace Microsoft.Data
                     throw new NotSupportedException();
 <# } else { #>
                     PrimitiveColumn<<#=type.TypeName#>> <#=type.TypeName#>Column = this as PrimitiveColumn<<#=type.TypeName#>>;
-                    var new<#=type.TypeName#>Column = <#=type.TypeName#>Column.Clone();
+                    PrimitiveColumn<<#=type.TypeName#>> new<#=type.TypeName#>Column;
+                    if (inPlace)
```
Comment (gfoidl) [inline src/Microsoft.Data/PrimitiveColumn.BinaryOperations.tt]: My previous suggestion should be applied here (I know you know this...).

File: tests/Microsoft.Data.DataFrame.Tests/DataFrameTests.cs [hunk 2]
```diff
@@ -541,17 +544,29 @@ public void TestComputations()
 
             // Test the computation results
             df["Double"][0] = 100.0;
-            df["Double"].CumulativeMax();
+            BaseColumn doubleColumn = df["Double"].CumulativeMax();
+            Assert.Equal(100.0, doubleColumn[9]);
```
Comment (eerhardt) [inline tests/Microsoft.Data.DataFrame.Tests/DataFrameTests.cs]: Should we be verifying more than just the 9th value?

File: src/Microsoft.Data/GroupBy.cs [hunk 3]
```diff
@@ -17,13 +17,13 @@ public abstract class GroupBy
         /// Compute the number of non-null values in each group 
         /// </summary>
         /// <returns></returns>
-        public abstract DataFrame Count();
+        public abstract DataFrame Count(params string[] columnNames);
```
Comment (eerhardt) [inline src/Microsoft.Data/GroupBy.cs]: These changes seem unrelated. Do we have tests covering this (passing in columnNames)?

Comment (pgovind) [reply src/Microsoft.Data/GroupBy.cs]: Yup. The last parts of TestGroupBy

**Comment by pgovind** [comment]
Thanks for the review folks!
